c-dependencies/js-compute-runtime/js-compute-builtins.cpp

```diff
@@ -694,11 +694,244 @@ bool parse_body(JSContext *cx, HandleObject self, UniqueChars buf, size_t len) {
   return JS::ResolvePromise(cx, result_promise, result);
 }
 
-bool consume_content_stream_for_bodyAll(JSContext *cx, HandleObject self, HandleObject stream,
-                                        HandleValue body_parser) {
-  JS_ReportErrorLatin1(cx, "Consuming a content-provided ReadableStream as a body using "
-                           ".text(), .json(), or .arrayBuffer() not yet supported");
-  return false;
+bool content_stream_read_then_handler(JSContext *cx, HandleObject self, HandleValue extra,
+                                      CallArgs args) {
+  RootedObject then_handler(cx, &args.callee());
+  // The reader is stored in the catch handler, which we need here as well.
+  // So we get that first, then the reader.
+  MOZ_ASSERT(extra.isObject());
+  RootedObject catch_handler(cx, &extra.toObject());
+#ifdef DEBUG
+  bool foundContents;
+  if (!JS_HasElement(cx, catch_handler, 1, &foundContents)) {
+    return false;
+  }
+  MOZ_ASSERT(foundContents);
+#endif
+  RootedValue contents_val(cx);
+  if (!JS_GetElement(cx, catch_handler, 1, &contents_val)) {
+    return false;
+  }
+  MOZ_ASSERT(contents_val.isObject());
+  RootedObject contents(cx, &contents_val.toObject());
+  if (!contents) {
+    return false;
+  }
+#ifdef DEBUG
+  bool contentsIsArray;
+  if (!JS::IsArrayObject(cx, contents, &contentsIsArray)) {
+    return false;
+  }
+  MOZ_ASSERT(contentsIsArray);
+#endif
+
+  auto reader_val = js::GetFunctionNativeReserved(catch_handler, 1);
+  MOZ_ASSERT(reader_val.isObject());
+  RootedObject reader(cx, &reader_val.toObject());
+
+  // TODO: Confirm this -- I think it is not true...
+  // We're guaranteed to work with a native ReadableStreamDefaultReader here,
+  // which in turn is guaranteed to vend {done: bool, value: any} objects to
```

Comment:
Is `vend` a typo?

Reply:
I'll change to `return` instead of `vend` as it seems less confusing 👍